### PR TITLE
drop_first attribute added in encode features

### LINKED ISSUE #635: Feature Request: Ability to drop redundant variable in ft.encode_features()
Is it possible to add a _drop_first_ parameter to the _ft.encode_features()_ function similar to _pandas.get_dummies()_ ?

### PR DESCRIPTION
**Pull request by ayushpatidar**
### Pull Request Description
drop_first attribute is added as param in encode_feature() function. 
This feature will help user to control on redundant feature in dataframe

Fixes #635
-----
*After creating the pull request: in order to pass the **changelog_updated** check you will need to update the "Future Release" section of* `docs/source/changelog.rst` *to include this pull request.*

### CODE REVIEW

**Comment by kmax12** [comment]
Thank you for the contribution, @ayushpatidar.

A couple things 

1. can you sign the CLA? (you may have to do it a second time for this PR). 
2. Please add a test case for this new parameter to https://github.com/Featuretools/featuretools/blob/master/featuretools/tests/synthesis/test_encode_features.py

Once you are passing all the github checks, we can review.

File: featuretools/synthesis/encode_features.py [hunk 1]
```diff
@@ -23,6 +23,9 @@ def encode_features(feature_matrix, features, top_n=10, include_unknown=True,
                 defaults to encode all necessary features.
             inplace (bool): Encode feature_matrix in place. Defaults to False.
             verbose (str): Print progress info.
+            drop_first (bool): Whether to get l-1 dummies out of l categorical
```
Comment (kmax12) [inline featuretools/synthesis/encode_features.py]: let's use `k` instead of `l`

File: featuretools/synthesis/encode_features.py [hunk 2]
```diff
@@ -113,7 +118,12 @@ def encode_features(feature_matrix, features, top_n=10, include_unknown=True,
         val_counts = val_counts.sort_values([f.get_name(), index_name],
                                             ascending=False)
         val_counts.set_index(index_name, inplace=True)
-        unique = val_counts.head(top_n).index.tolist()
+        select_n = top_n
+        if drop_first:
+            if len(val_counts) != 1 or top_n != 1:
```
Comment (kmax12) [inline featuretools/synthesis/encode_features.py]: what if we remove the conditional and just do these two lines?

```
select_n = min(len(val_counts), top_n) 
select_n = max(select_n - 1, 1)  # make sure at least 1 category is selected
```

File: featuretools/tests/synthesis/test_encode_features.py [hunk 3]
```diff
@@ -139,3 +139,18 @@ def test_encode_features_topn(es):
...
         assert features_enc.columns.tolist().count(name) == 1
+
+
+def test_encode_features_drop_first():
+    df = pd.DataFrame({'category': ['ao', 'b', 'c', 'd', 'e']})
+    es = EntitySet('test')
+    es.entity_from_dataframe(entity_id='a', dataframe=df, index='index', make_index=True)
+    features, feature_defs = dfs(entityset=es, target_entity='a')
+    features_enc, feature_defs_enc = encode_features(features, feature_defs,
+                                                     drop_first=True, top_n=10, include_unknown=False)
```
Comment (kmax12) [inline featuretools/tests/synthesis/test_encode_features.py]: no need to include `top_n` in this test case

File: featuretools/synthesis/encode_features.py (hunk 2 above)
Comment (ayushpatidar) [reply featuretools/synthesis/encode_features.py]: @kmax12 
But we have to keep outer conditional statement ( if drop_first ).

Comment (kmax12) [reply featuretools/synthesis/encode_features.py]: yep, sorry, I was just referring to inner conditional

**Pull request merged**